engines/wasmtime/wasmtime.go

```diff
@@ -88,10 +89,16 @@ func Engine() wapc.Engine {
 // on wapc.Engine is called. The result is closed upon wapc.Module Close.
 type NewRuntime func() (*wasmtime.Engine, error)
 
+type CacheImpl func(*wasmtime.Engine, []byte) (*wasmtime.Module, error)
+
 // EngineWithRuntime allows you to customize or return an alternative to
 // DefaultRuntime,
-func EngineWithRuntime(newRuntime NewRuntime) wapc.Engine {
-	return &engine{newRuntime: newRuntime}
+func EngineWithRuntime(newRuntime NewRuntime, caching ...CacheImpl) wapc.Engine {
```

Comment:
if we end up going this way, where implementation details aren't actually defined by the runtime, I would suggest adding an options type or similar. That way the next thing you need doesn't break this signature.

Reply:
I wasn't sure the best way to conceal the detail and not be a breaking change.  Let me look over that again.

Reply:
So, specifically I meant use varargs with an options pattern like this: https://github.com/http-wasm/http-wasm-host-go/blob/main/handler/options.go https://github.com/http-wasm/http-wasm-host-go/blob/main/handler/nethttp/middleware.go#L22

Reply:
Is what I pushed acceptable?  This gives full flexibility unless I am missing something.